src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/ISpeechToText.shared.cs

```diff
@@ -28,32 +28,42 @@ public interface ISpeechToText : IAsyncDisposable
 	SpeechToTextState CurrentState { get; }
 
 	/// <summary>
-	/// Converts speech to text in real time.
+	/// Starts the SpeechToText service
 	/// </summary>
+	/// <remarks>
+	/// Real time speech recognition results will be surfaced via <see cref="RecognitionResultUpdated"/> and <see cref="RecognitionResultCompleted"/>
+	/// </remarks>
 	/// <param name="culture">Speak language</param>
-	/// <param name="recognitionResult">Intermediate recognition result.</param>
 	/// <param name="cancellationToken"><see cref="CancellationToken"/></param>
-	/// <returns>Final recognition result</returns>
-	Task<SpeechToTextResult> ListenAsync(CultureInfo culture, IProgress<string>? recognitionResult, CancellationToken cancellationToken = default);
+	Task StartListenAsync(CultureInfo culture, CancellationToken cancellationToken = default);
 
 	/// <summary>
-	/// Starts the SpeechToText service
+	/// Stops the SpeechToText service
+	/// </summary>
+	/// <remarks>
+	/// Speech recognition results will be surfaced via <see cref="RecognitionResultCompleted"/>
+	/// </remarks>
+	/// <param name="cancellationToken"><see cref="CancellationToken"/></param>
+	Task StopListenAsync(CancellationToken cancellationToken = default);
+	
+	/// <summary>
+	/// Starts the Offline SpeechToText service
 	/// </summary>
 	/// <remarks>
 	/// Real time speech recognition results will be surfaced via <see cref="RecognitionResultUpdated"/> and <see cref="RecognitionResultCompleted"/>
 	/// </remarks>
 	/// <param name="culture">Speak language</param>
 	/// <param name="cancellationToken"><see cref="CancellationToken"/></param>
-	Task StartListenAsync(CultureInfo culture, CancellationToken cancellationToken = default);
+	Task StartOfflineListenAsync(CultureInfo culture, CancellationToken cancellationToken = default);
```

Comment:
Do you think there is a way that we could expose this via a different interface? Then we can keep the related methods together and separate from unrelated methods

Reply:
It's a great idea, I think we should keep only 2 methods here (Start/Stop), but implement the interface with Offline and Online Speech to Text services. We can even provide a default strategy to choose Online or Offline implementation based on internet connectivity.
Let's bring this discussion to our stand up.

Reply:
Yes perfect! I think that should provide a nice API surface for developers then